src/app/shared/services/user-session.service.ts

```diff
@@ -17,35 +18,46 @@ export interface UserSession {
 export class UserSessionService implements OnDestroy {
 
   session$ = new BehaviorSubject<UserSession|undefined>(undefined)
+  userProfileError$ = new Subject<void>()
+  watchedGroup$ = this.session$.pipe(map(session => session?.watchedGroup), distinctUntilChanged())
```

Comment:
`distinctUntilChanged` is probably not able to deal properly with a "Group". Maybe that's not very important if the only goal is to get rid of the multiple `undefined` though. (maybe should be written in comments at least)

Reply:
Fixed